scripts/parse-coverage-output.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Parses the output of "llvm-cov report" and reformats it."""
+
+import sys
+
+def split_columns(line):
+  """Splits the tokens separated by two or more spaces.
+
+  This is the format of the "table" that's emitted by llvm-cov report.
+  """
+  return [item.strip() for item in line.split('  ') if item]
+
+def parse_llvm_cov_report_from_stdin():
+  """Interprets the whole stdin as a "table" from llvm-cov report.
+
+  Returns:
+    (column_names, rows)
+  """
+  # Parse the table's header.
+  header_line = next(sys.stdin)
+  column_names = split_columns(header_line)
+  if len(column_names) <= 1 or column_names[0] != 'Filename':
+    raise RuntimeError(f'Unexpected header: ${header_line}')
+  # Parse the table's contents.
+  column_names = split_columns(header_line)
+  rows = []
+  for line in sys.stdin:
+    cells = split_columns(line)
+    if len(cells) <= 1:
+      # Empty line, or line with separators ("---..."), or line with a subtitle
+      # (e.g., "Files which contain no functions:").
+      continue
+    if len(cells) != len(column_names):
+      raise RuntimeError(f'Unexpected row line: ${line}')
+    rows.append(cells)
```

Comment:
Looks good, but do we actually need to parse all of this if we only look at the last (non-separator etc.) line anyway?

Reply:
I was planning to print the whole table in a follow-up commit.

The reason I parse it (and not just post the whole llvm-cov's output) is that this textual format is really hard to read because of long lines. Here's how it looks like: https://pastebin.com/yFajjvni
If I reformat the same data into a Markdown table it should be more readable (we can omit some columns if we want).